### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by emmenlau**
This PR adds a number of improvement to the cmake build of LibRaw. I've split it into several commits to make the review of each commit easier and the changes within each commit consistent.

Please let me know what you think?

### CODE REVIEW

**Comment by letmaik** [comment]
Thanks for the work, looks good in general. Windows CI is failing during linking. Any idea?

**Comment by emmenlau** [comment]
> Thanks for the work, looks good in general. Windows CI is failing during linking. Any idea?

Thanks for the thumbs up! It seems the linker error comes from missing `htonl` that is referenced in some form from the tiff integration. Linking `Ws2_32` on MSVC should help. I've added the change, lets see...

File: CMakeLists.txt [hunk 1]
```diff
@@ -472,47 +462,55 @@ add_library(raw ${libraw_LIB_SRCS})
...
+if (WIN32)
+    target_compile_definitions(raw PRIVATE LIBRAW_BUILDLIB)
+endif()
+
+# Static builds use LIBRAW_NODLL:
+if(NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(raw PUBLIC LIBRAW_NODLL)
+endif()
+
 # Disable compilation warnings from LibRaw. Just to be clear on the console.
```
Comment (letmaik) [inline CMakeLists.txt]: Let's remove that comment.

File: CMakeLists.txt [hunk 2]
```diff
@@ -472,47 +462,55 @@ add_library(raw ${libraw_LIB_SRCS})
...
 # Use O4 GCC compilation option to prevent artifacts with OpenMP
-IF(WIN32 AND MSVC)
-    target_compile_options(raw PRIVATE -w)
-ELSE()
-    target_compile_options(raw PRIVATE -w -O4)
+IF(NOT MSVC)
+    target_compile_options(raw PRIVATE -O3)
```
Comment (letmaik) [inline CMakeLists.txt]: Let's remove this altogether. Optimization flags are set via CMAKE_BUILD_TYPE anyway and I don't see why there has to be some special gcc override.

File: CMakeLists.txt [hunk 3]
```diff
@@ -524,13 +522,21 @@ SET_TARGET_PROPERTIES(raw PROPERTIES COMPILE_PDB_NAME "raw")
...
+if (WIN32)
+    target_compile_definitions(raw_r PRIVATE LIBRAW_BUILDLIB)
+endif()
+
+# Static builds use LIBRAW_NODLL:
+if(NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(raw_r PUBLIC LIBRAW_NODLL)
+endif()
+
+IF(NOT MSVC)
```
Comment (letmaik) [inline CMakeLists.txt]: See comment above.

**Review by letmaik — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (emmenlau) [reply CMakeLists.txt]: Sounds good, done.

File: CMakeLists.txt (hunk 2 above)
Comment (emmenlau) [reply CMakeLists.txt]: Sounds good, done.

File: CMakeLists.txt (hunk 3 above)
Comment (emmenlau) [reply CMakeLists.txt]: Sounds good, done.

**Comment by emmenlau** [comment]
I've just converted all uppercase cmake command names consistently to lowercase (this was mixed in the file). I hope that's good?

**Pull request merged**